test/main.spec.ts

```diff
@@ -67,37 +57,43 @@ test('meta tags are added and removed', async ({ page }) => {
 
   const extraHome = page.locator('[name="extra"]');
   await expect(extraHome).toHaveCount(0);
-
-
 });
 
-
-
 test('prefetching works', async ({ page }) => {
-
   await page.goto('http://localhost:3000');
   let preAbout = page.locator('link[href="/about"]');
-  await expect(preAbout).toHaveCount(1)
+  await expect(preAbout).toHaveCount(1);
 
   const about = page.locator('#about');
   await about.click();
 
   const homePre = page.locator('link[href="/"]');
-  await expect(homePre).toHaveCount(1)
+  await expect(homePre).toHaveCount(1);
 
   let testPre = page.locator('link[href="/test"]');
-  await expect(testPre).toHaveCount(0)
+  await expect(testPre).toHaveCount(0);
 
   // Validate intersection observer works
   const heading = page.locator('#chapter');
   await heading.click();
 
   testPre = page.locator('link[href="/test"]');
-  await expect(testPre).toHaveCount(1)
+  await expect(testPre).toHaveCount(1);
 
   // ensure no duplicates
   preAbout = page.locator('link[href="/about"]');
-  await expect(preAbout).toHaveCount(1)
+  await expect(preAbout).toHaveCount(1);
+});
 
+test('page change announcer works', async ({ page }) => {
+  await page.goto('http://localhost:3000');
 
-});
\ No newline at end of file
+  const announcerHome = page.locator('#flamethrower-announcer');
+  await expect(announcerHome).toHaveText('Flamethrower');
+
+  const about = page.locator('#about');
+  await about.click();
+
+  const announcerAbout = page.locator('#flamethrower-announcer');
+  await expect(announcerAbout).toHaveText('About');
```

Comment:
FYI, there is a tool for validating these announcements. https://www.npmjs.com/package/extend-to-be-announced
Though I'm not sure whether it integrates with playwright nicely.

Reply:
Nice, I didn't know about this one. Although, not sure if it is a good idea add a new dev dependency only for this test case 🤔.

Another way of optimizing the test that I was thinking is:

I could use the same variable `const announcer = page.locator('#flamethrower-announcer');` for `home` page and `about` page as technically is the same node, so maybe it keeps the reference, but I didn't try it and I don't know if it works